### PR TITLE
[ClangCL] fix warning in DxbcConverter

### PR DESCRIPTION
**Pull request by python3kgae**
1. fix order of fields in constructor.
2. fix compare warning for type mismatch by cast to same type.
3. use {} instead of {0} when initialize to all zero.
4. add override for override methods.

### CODE REVIEW

**Review by pow2clk — COMMENTED** [review]
Just a couple questions

File: projects/dxilconv/lib/DxbcConverter/DxbcConverter.cpp [hunk 1]
```diff
@@ -802,7 +802,7 @@ void DxbcConverter::ExtractSignatureFromDDI(
     } else {
       E.SemanticName = string(DXBC::GetD3D10SBName(P.SystemValue));
       E.SemanticIndex = DXBC::GetD3D10SBSemanticIndex(P.SystemValue);
-      if (P.RegisterComponentType != D3D_REGISTER_COMPONENT_UNKNOWN) {
+      if (P.RegisterComponentType != D3D10_SB_REGISTER_COMPONENT_UNKNOWN) {
```
[the remark this answers is not in the document]
Comment (python3kgae) [reply projects/dxilconv/lib/DxbcConverter/DxbcConverter.cpp:805]: The cast should be safe. The value for those enums are matching.

File: include/dxc/DxilContainer/DxilContainerReader.h [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 namespace hlsl {
 
-#define DXIL_CONTAINER_BLOB_NOT_FOUND -1
+const uint32_t DXIL_CONTAINER_BLOB_NOT_FOUND = -1;
```
Comment (damyanp) [inline include/dxc/DxilContainer/DxilContainerReader.h:21]: Doesn't this generate a warning with MSVC?  Or have we disabled that warning (eek)?

```
> type t.cpp ; cl /c /Wall t.cpp
#include <stdint.h>

const uint32_t foo = -1;
Microsoft (R) C/C++ Optimizing Compiler Version 19.37.32824 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

t.cpp
t.cpp(3): warning C4245: 'initializing': conversion from 'int' to 'uint32_t', signed/unsigned mismatch
t.cpp(3): warning C5264: 'foo': 'const' variable is not used
```

**Review by damyanp — CHANGES_REQUESTED** [review]

**Review by pow2clk — APPROVED** [review]
Looks good

**Pull request merged**